lib/src/token.dart

```diff
@@ -25,55 +25,56 @@ class StartTagToken extends TagToken {
   LinkedHashMap<Object, String> data;
 
   /// The attribute spans if requested. Otherwise null.
-  List<TagAttribute> attributeSpans;
+  List<TagAttribute>? attributeSpans;
 
   bool selfClosingAcknowledged;
 
   /// The namespace. This is filled in later during tree building.
-  String namespace;
+  String? namespace;
 
-  StartTagToken(String name,
-      {this.data,
+  StartTagToken(String? name,
+      {LinkedHashMap<Object, String>? data,
```

Comment:
Can this not just be `this.data = const {}`? I don't see it being modified right here but could be wrong.

Reply:
Unfortunately not because it is public as a `LinkedHashMap` and not a `Map`. I'm not sure if that's critical, but it does mean that we can't assign `{}` without having to also cast it.

Reply:
ah ok, this seems fine then